api/notd/gallery_manager.py

```diff
@@ -111,22 +111,24 @@ async def get_gallery_token(self, registryAddress: str, tokenId: str) -> Gallery
             return galleryToken
         if collection.doesSupportErc1155:
             query = (
-                sqlalchemy.select(TokenMetadatasTable, TokenCustomizationsTable, LatestTokenListingsTable)
+                sqlalchemy.select(TokenMetadatasTable, sqlalchemy.func.sum(TokenMultiOwnershipsTable.c.quantity))
                     .join(TokenCustomizationsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenCustomizationsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenCustomizationsTable.c.tokenId), isouter=True)
                     .join(TokenMultiOwnershipsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == TokenMultiOwnershipsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == TokenMultiOwnershipsTable.c.tokenId))
-                    .join(LatestTokenListingsTable, sqlalchemy.and_(TokenMetadatasTable.c.registryAddress == LatestTokenListingsTable.c.registryAddress, TokenMetadatasTable.c.tokenId == LatestTokenListingsTable.c.tokenId, LatestTokenListingsTable.c.offererAddress == TokenMultiOwnershipsTable.c.ownerAddress, LatestTokenListingsTable.c.endDate >= datetime.datetime.now()), isouter=True)
                     .where(TokenMetadatasTable.c.registryAddress == registryAddress)
                     .where(TokenMetadatasTable.c.tokenId == tokenId)
-                    .order_by(sqlalchemy.func.coalesce(LatestTokenListingsTable.c.value, 0).asc())
+                    .group_by(TokenMetadatasTable.c.tokenMetadataId, TokenMetadatasTable.c.registryAddress, TokenMetadatasTable.c.tokenId)
             )
             result = await self.retriever.database.execute(query=query)
             row = result.first()
             if not row:
                 raise NotFoundException(message=f'GalleryToken with registry:{registryAddress} tokenId:{tokenId} not found')
+            print('here')
             galleryToken = GalleryToken(
                 tokenMetadata=token_metadata_from_row(row),
-                tokenCustomization=token_customization_from_row(row) if row[TokenCustomizationsTable.c.tokenCustomizationId] else None,
-                tokenListing=token_listing_from_row(row) if row[LatestTokenListingsTable.c.latestTokenListingId] else None,
+                tokenCustomization=token_customization_from_row(row) if collection.doesSupportErc721 else None,
+                tokenListing=token_listing_from_row(row) if collection.doesSupportErc721 else None,
+                quantity=row[-1] if collection.doesSupportErc1155 else 1
```

Comment:
you should label the field and use the label, not -1. this is done somewhere in this file, search for "label"

Reply:
also you can just add a constant into the erc721 branch to make this slightly cleaner

Reply:
*add a constant into the sql like `select ..., 1 as quantity`